src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -11,6 +11,643 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     [CompilerTrait(CompilerFeature.ReadOnlyReferences)]
     public class CodeGenInParametersTests : CompilingTestBase
     {
+        [Fact]
+        public void ThreeParamReorder()
+        {
+            var comp = CompileAndVerify(@"
+using System;
+class C
+{
+    public struct S
+    {
+        public int X;
+    }
+
+    private static S _field;
+
+    public static ref S GetField(int order)
+    {
+        Console.WriteLine(""GetField "" + _field.X++ + "" "" + order);
+        return ref _field;
+    }
+
+    public static void Main()
+    {
+        M(y: in GetField(0).X, z: GetField(1).X, x: GetField(2).X);
+    }
+
+    static void M(in int x, in int y, in int z)
+    {
+        Console.WriteLine(x);
+        Console.WriteLine(y);
+        Console.WriteLine(z);
+    }
+}", expectedOutput: @"GetField 0 0
+GetField 1 1
+GetField 2 2
+3
+3
+3");
+            comp.VerifyIL("C.Main", @"
+{
+  // Code size       43 (0x2b)
+  .maxstack  3
+  .locals init (int& V_0,
+                int& V_1)
+  IL_0000:  ldc.i4.0
+  IL_0001:  call       ""ref C.S C.GetField(int)""
+  IL_0006:  ldflda     ""int C.S.X""
+  IL_000b:  stloc.0
+  IL_000c:  ldc.i4.1
+  IL_000d:  call       ""ref C.S C.GetField(int)""
+  IL_0012:  ldflda     ""int C.S.X""
+  IL_0017:  stloc.1
+  IL_0018:  ldc.i4.2
+  IL_0019:  call       ""ref C.S C.GetField(int)""
+  IL_001e:  ldflda     ""int C.S.X""
+  IL_0023:  ldloc.0
+  IL_0024:  ldloc.1
+  IL_0025:  call       ""void C.M(in int, in int, in int)""
+  IL_002a:  ret
+}");
+
+        }
+
+        [Fact]
+        public void InParamReadonlyFieldReorder()
+        {
+            var comp = CompileAndVerify(@"
+using System;
+class C
+{
+    private static readonly int _f = 0;
+    public C()
+    {
+        M(y: _f, x: _f + 1);
+    }
+
+    public static void Main()
+    {
+        M(y: _f, x: _f + 1);
```

Comment:
>M(y: _f, x: _f + 1); [](start = 8, length = 20)

Would it be interesting to test this case with explicit ```in``` modifiers at the call site?